coreax/data.py

```diff
@@ -207,19 +207,23 @@ class Data(eqx.Module):
         the ones vector (implies a scalar weight of one);
     """
 
-    data: Shaped[Array, " n d"]
+    data: Shaped[Array, " n *d"]
     weights: Shaped[Array, " n"]
 
     def __init__(
         self,
-        data: Shaped[ArrayLike, " n d"],
+        data: Shaped[ArrayLike, " n *d"],
         weights: Shaped[ArrayLike, " n"] | None = None,
     ):
         """Initialise Data class."""
-        self.data = jnp.atleast_2d(data)
-        n = self.data.shape[0]
+        self.data = jnp.asarray(data)
```

Comment:
Same comment about `jnp.asarray` vs `jnp.atleast_2d` - is this needed and has the code been tested if 1d arrays are passed?

Reply:
@tc85324 this needs a `atleast_2d` wrapper I believe

Reply:
The type hint indicates that d is optional, so will be treated as an n-vector. In this case we have logical consistency as weights is expected to be an n-vector and the docstrings make clear that each weight is associated with a leading index of the data. If you have atleast_2d here, you end up with the confusing scenario where you pass an `n-vector` of data but are only expected to pass a single scalar weight. In addition, there is no reason why data should have to be 2d here, whereas at the point of use (in the gramian_row_mean of the approximators) the mathematical operations doe require 2d.